src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -87,6 +88,234 @@ protected override Conversion GetInterpolatedStringConversion(BoundUnconvertedIn
+                    return false;
+                }
+
+                var argsToParam = overloadResolutionResult.ValidResult.Result.ArgsToParamsOpt;
+
+                _binder.CoerceArguments(overloadResolutionResult.ValidResult, analyzedArguments.Arguments, diagnostics);
+
+                bool expanded = overloadResolutionResult.ValidResult.Result.Kind == MemberResolutionKind.ApplicableInExpandedForm;
+                _binder.BindDefaultArguments(
+                    source.Syntax,
+                    overloadResolutionResult.ValidResult.Member.Parameters,
+                    analyzedArguments.Arguments,
+                    analyzedArguments.RefKinds,
+                    ref argsToParam,
+                    out BitVector defaultArguments,
+                    expanded,
+                    enableCallerInfo: true,
+                    diagnostics);
+
+                builderFormatCalls.Add(new MethodArgumentInfo(overloadResolutionResult.ValidResult.Member, analyzedArguments.Arguments.ToImmutable(), argsToParam, defaultArguments, expanded));
+            }
+
+            free();
+            diagnostics.AddDiagnostics(source.Syntax, useSiteInfo);
+            builderArguments = builderFormatCalls.ToImmutableAndFree();
+            interpolationHoleCandidates?.Free();
+            return true;
+
+            bool tryGetCandidateMethods(string methodName, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo, [NotNullWhen(true)] out ArrayBuilder<MethodSymbol>? candidateMethods)
+            {
+                var lookupResult = LookupResult.GetInstance();
+                _binder.LookupSymbolsSimpleName(lookupResult,
+                    builderType,
+                    methodName,
+                    arity: 0,
+                    basesBeingResolved: null,
+                    options: LookupOptions.AllMethodsOnArityZero | LookupOptions.MustBeInstance | LookupOptions.MustBeInvocableIfMember,
+                    diagnose: true,
+                    ref useSiteInfo);
+
+                if (!lookupResult.IsMultiViable)
+                {
+                    candidateMethods = null;
+                    lookupResult.Free();
+                    return false;
+                }
+
+                Debug.Assert(lookupResult.Symbols.Count > 0);
+                candidateMethods = ArrayBuilder<MethodSymbol>.GetInstance(lookupResult.Symbols.Count);
+
+                foreach (var symbol in lookupResult.Symbols)
+                {
+                    if (symbol is not MethodSymbol method)
+                    {
+                        // PROTOTYPE(interp-strings): We'll want to have a specific error for when we're converting to a non-string type
+                        candidateMethods.Free();
+                        lookupResult.Free();
+                        candidateMethods = null;
+                        return false;
+                    }
+
+                    candidateMethods.Add(method);
```

Comment:
>method [](start = 41, length = 6)

Can we get an extension method here?

Reply:
Shouldn't be possible, we have `LookupOptions.MustBeInstance` without `LookupOptions.IncludeExtensionMethods`.

---
In reply to: [610747594](https://github.com/dotnet/roslyn/pull/52422#discussion_r610747594) [](ancestors = 610747594)